parsetag.go

```diff
@@ -137,46 +137,70 @@ func (tag Tag) Fill(model interface{}, opts ...FillOptArg) error {
 	if !v.IsValid() || v.IsNil() || v.Type().Kind() != reflect.Ptr || v.Type().Elem().Kind() != reflect.Struct {
 		return errors.Errorf("Fill target must be a pointer to a struct, not %T", model)
 	}
+
+	elements := strings.Split(tag.Value, ",")
+
+	// Now walk over the input model that controls the parsing.
+	// 1.first round match by index
+	elementsConsumed := make([]bool, len(elements))
+	err := WalkStructElementsWithError(v.Type(), func(f reflect.StructField) error {
+		parts, err := getTagContents(f.Tag, opt.tag)
+		if err != nil {
+			return err
+		}
+		var value string
+		if len(parts) > 0 && parts[0] != "" {
+			i, err := strconv.Atoi(parts[0])
+			if err == nil {
+				// positional!
+				if i >= len(elements) {
+					return nil
+				}
+				value = elements[i]
+				elementsConsumed[i] = true
+			}
+		}
+		return setValueHelper(value, parts, f, v)
+	})
+	if err != nil {
+		return err
+	}
+	// 2.remove matched by index
+	newElements := []string{}
+	for i, e := range elements {
+		if elementsConsumed[i] {
+			continue
+		}
+		newElements = append(newElements, e)
+	}
```

Comment:
```suggestion
	newElements := []string{}
	for i, e := range elements {
		if elementsConsumed[i] {
			continue
		}
		newElements = append(newElements, e)
	}
	elements = newElements
```

just use this to filter elements, you don't need a new variable exposed to the rest of the function.